### ISSUE TITLE
Accessible, discoverable tooltips

### ISSUE THREAD

**Issue by amyjko** [wordplaydev/wordplay#536]
### What happened?

At the CSTA Wordplay workshop, some teachers on some devices reported problems with tooltips:

* Not knowing they existed
* Not showing after a dwell
* Not showing until a long dwell
* Font sizes being too small to read on their device
* Not being available on touch screens

All of these are well know problems with tooltips. We need to think about better ways of exposing the tooltip information that is available. There are many possible solutions, and a long history of help interface alternatives. Choose one that is consistent across platforms, but also ARIA compliant.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What operating system are you using?

Windows, macOS

**Comment by amyjko** [comment]
The default tooltip pattern with the `title` attribute is generally inaccessible. Web standards have been slow to address this, but there is progress, especially with the [popover](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API/Using) API, which now has 90% global usage support. There's also no settled [WCAG tooltip pattern yet](https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/) yet, so it's up to use to find a usable and accessible pattern.

Our general requirements are going to be:

- [ ] Tooltips are plain text
- [ ] Tooltips should be automatically shown when a component with a tooltip receives focus or is hovered after a period of time
- [ ] The hover delay should be customizable
- [ ] Tooltips should be dismissible either by pressing a escape or clicking a button on the tooltip to dismiss it.
- [ ] Tooltips should be displayable in one of 8 locations relative to a component, to minimize obscuring other content. This should be a required attribute

From an implementation perspective, I think we should:

- [ ] Use the `popover` API and the `hint` state
- [ ] Create a tooltip component that encapsulates all its functionality
- [ ] For all components that accept a `title` attribute, include a tooltip component instead of the title (which generally [shouldn't be used](https://www.a11yproject.com/posts/title-attributes/)).
